src/main/kotlin/org/rust/lang/core/psi/ext/PsiElement.kt

```diff
@@ -332,3 +332,20 @@ fun PsiElement.isKeywordLike(): Boolean {
         else -> false
     }
 }
+
+/**
+ * Consider we do some `resolve` in a quick-fix which is called in preview mode.
+ * Quick-fix is called on copy of the original file, but `resolve` will return original element.
+ * We will have an exception if we try to modify the original element.
+ * Thus, we should call this function on `resolve` result to obtain element in the copy of the original file.
+ */
+fun <T: PsiElement> T.findPreviewCopy(startElement: PsiElement): T? {
```

Comment:
What do you think about passing `copyFile` directly instead of `startElement`? Currently, it's not easy to understand what `startElement` is since you use it only to get its `containingFile`. At the same time, in some cases you already have a copy file, for example, in `com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement#invoke(com.intellij.openapi.project.Project, com.intellij.psi.PsiFile, com.intellij.openapi.editor.Editor, com.intellij.psi.PsiElement, com.intellij.psi.PsiElement)`